src/catalog/catalog_entry/sequence_catalog_entry.cpp

```diff
@@ -0,0 +1,123 @@
+#include "catalog/catalog_entry/sequence_catalog_entry.h"
+
+#include "binder/ddl/bound_create_sequence_info.h"
+#include "common/exception/catalog.h"
+#include "common/exception/overflow.h"
+#include "function/arithmetic/add.h"
+
+using namespace kuzu::binder;
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace catalog {
+
+SequenceData SequenceCatalogEntry::getSequenceData() {
+    std::lock_guard<std::mutex> lck(mtx);
+    return sequenceData;
+}
+
+int64_t SequenceCatalogEntry::currVal() {
+    std::lock_guard<std::mutex> lck(mtx);
+    int64_t result;
+    if (sequenceData.usageCount == 0u) {
+        throw CatalogException("curr_val: sequence \"" + name + "\" is not yet defined");
```

Comment:
Let's be more user-friendly here. Can u add one more sentence along the line of "please try `next_val()` first before calling `curr_val()`"? Just that users can directly get the idea and fix it without consulting our docs.

Reply:
sounds good!